applications/pelion_client/CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+cmake_minimum_required(VERSION 3.13.1)
+
+
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE ZDebug)
+endif()
```

Comment:
Please use the Zephyr build types instead.
https://docs.zephyrproject.org/latest/guides/kconfig/setting.html#the-initial-configuration

This means you can create:
- `prj.conf`: The default config file. Corresponding to what you have in `ZDebug.conf` today.
- `prj_mbedtls.conf`: The mbedtls configuration. Corresponding to what you have in `ZDebugMbedTLS.conf` today..
- `prj_release.conf`: This could be a release config.

The advantage of this approach is that it follows Zephyr build system, and that you will get corresponding board adjustment specified, for example building:
- `west build -bnrf52840dk_nrf52840` would include `boards/nrf52840dk_nrf52840.conf`
- `west build -bnrf52840dk_nrf52840 -- -DCONF_FILE=prj_mbedtls.conf` would include `boards/nrf52840dk_nrf52840_mbedtls.conf` instead.

Reply:
This won't work for us. I can discuss this offline in detail but in short this architecture allows to have separate caf configuration files for various boards.
Similarly as is at the moment done for applications/machine_learning and applications/nrf_desktop. 
If you find it a blocker it will be a serious issue that will impact the project.

Reply:
accepted as-is for current time.
Let's continue discussions on way forward, and adopt all apps to the final decision.